bundles/org.openhab.core.updater/src/main/java/org/openhab/core/updater/enums/VersionType.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.updater.enums;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link VersionType} determines what type of update to apply
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public enum VersionType {
+    STABLE("stable"),
+    MILESTONE("unstable"),
+    SNAPSHOT("testing"),
```

Comment:
Milestone and snapshot repo should be the other way around, see https://www.openhab.org/docs/installation/linux.html#apt-based-systems

Reply:
Hi @pacive many thanks for the review. You are right. Thanks for the link too, because it also tells me how to write the scripts for yum and dnf based package managers. I will do both of those in a forthcoming commit, although I will need to ask others to test them, since I don’t own such systems myself..

Reply:
I hope you manage to make this work for all different OSs, this would be a great addition to OpenHAB IMO (even though personally I'd probably continue to upgrade via apt directly)

Reply:
> hope you manage to make this work for all different OSs

Hi @pacive the module now has classes that are either a) fully coded and tested, b) initial code and not tested, or c) stub code only, for the following Operating Systems, Flavours, and Package Managers..
- Windows (tested)
- Linux / Debian / Apt (tested)
- Linux / RedHat / Yum (initial code, not tested)
- Linux / Arch / PacMan (stub code)
- Linux / Gentoo / Portage (stub code)
- Mac (initial code, not tested)

Reply:
PS I fixed the mixup between UNSTABLE and TESTING too, so I will Resolve this conversation